### PR TITLE
[zh] Import sidebars from en-US

### PR DESCRIPTION
**Pull request by caugner**
### Description

Adds sidebar macro calls from the en-US version.

### Motivation

Sidebars have been added to almost all en-US pages, and many translated pages don't have them yet.

### Additional details

This was created using a custom version of the sync-translated-content tool (https://github.com/mdn/yari/commit/86ba2eac8690f3b8c1bca3cf5ad1aa813be4285c).

### CODE REVIEW

File: files/zh-cn/learn/html/howto/add_a_hit_map_on_top_of_an_image/index.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ title: Add a hit map on top of an image
 slug: Learn/HTML/Howto/Add_a_hit_map_on_top_of_an_image
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Learn/HTML/Howto")}}
```
Comment (yin1999) [inline files/zh-cn/learn/html/howto/add_a_hit_map_on_top_of_an_image/index.md:6]: This macro should use the localized path :)

See the screenshot:

image

```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Learn/HTML/Howto")}}
```

File: files/zh-cn/web/media/autoplay_guide/index.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ title: 媒体和 Web Audio API 的自动播放指南
 slug: Web/Media/Autoplay_guide
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Web/Media")}}
```
Comment (yin1999) [inline files/zh-cn/web/media/autoplay_guide/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Media")}}
```

File: files/zh-cn/web/media/dash_adaptive_streaming_for_html_5_video/index.md [hunk 3]
```diff
@@ -3,6 +3,8 @@ title: 为 HTML 5 视频提供的 DASH 自适应串流
 slug: Web/Media/DASH_Adaptive_Streaming_for_HTML_5_Video
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Web/Media")}}
```
Comment (yin1999) [inline files/zh-cn/web/media/dash_adaptive_streaming_for_html_5_video/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Media")}}
```

File: files/zh-cn/web/performance/critical_rendering_path/index.md [hunk 4]
```diff
@@ -3,6 +3,8 @@ title: 关键渲染路径
 slug: Web/Performance/Critical_rendering_path
 ---
 
+{{QuickLinksWithSubPages("Web/Performance")}}
```
Comment (yin1999) [inline files/zh-cn/web/performance/critical_rendering_path/index.md:6]: No items have been listed if only passed the slug without the prefix `/zh-CN/docs/`

A related comment: https://github.com/mdn/yari/issues/7438#issuecomment-1371880847

File: files/zh-cn/web/performance/optimizing_startup_performance/index.md [hunk 5]
```diff
@@ -3,6 +3,8 @@ title: 优化启动性能
 slug: Web/Performance/Optimizing_startup_performance
 ---
 
+{{QuickLinksWithSubPages("Web/Performance")}}
```
Comment (yin1999) [inline files/zh-cn/web/performance/optimizing_startup_performance/index.md:6]: +1

File: files/zh-tw/learn/common_questions/web_mechanics/what_is_a_web_server/index.md [hunk 6]
```diff
@@ -3,6 +3,8 @@ title: 何謂網路伺服器？
 slug: Learn/Common_questions/Web_mechanics/What_is_a_web_server
 ---
 
+{{QuicklinksWithSubPages("Learn/Common_questions")}}
```
Comment (yin1999) [inline files/zh-tw/learn/common_questions/web_mechanics/what_is_a_web_server/index.md:6]: +1

File: files/zh-tw/web/manifest/index.md [hunk 7]
```diff
@@ -3,7 +3,7 @@ title: Web App Manifest
 slug: Web/Manifest
 ---
 
-{{SeeCompatTable}}
+{{QuickLinksWithSubpages("/en-US/docs/Web/Manifest")}}{{SeeCompatTable}}
```
Comment (yin1999) [inline files/zh-tw/web/manifest/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Manifest")}}{{SeeCompatTable}}
```

File: files/zh-tw/web/media/index.md [hunk 8]
```diff
@@ -3,6 +3,8 @@ title: Web media technologies
 slug: Web/Media
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Web/Media")}}
```
Comment (yin1999) [inline files/zh-tw/web/media/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Media")}}
```

File: files/zh-tw/web/security/mixed_content/how_to_fix_website_with_mixed_content/index.md [hunk 9]
```diff
@@ -3,6 +3,8 @@ title: 如何修正含有混合內容的網站
 slug: Web/Security/Mixed_content/How_to_fix_website_with_mixed_content
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Web/Security")}}
```
Comment (yin1999) [inline files/zh-tw/web/security/mixed_content/how_to_fix_website_with_mixed_content/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Security")}}
```

File: files/zh-tw/web/security/mixed_content/index.md [hunk 10]
```diff
@@ -3,6 +3,8 @@ title: 混合內容
 slug: Web/Security/Mixed_content
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Web/Security")}}
```
Comment (yin1999) [inline files/zh-tw/web/security/mixed_content/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Security")}}
```

File: files/zh-tw/web/security/same-origin_policy/index.md [hunk 11]
```diff
@@ -3,6 +3,8 @@ title: 同源政策 (Same-origin policy)
 slug: Web/Security/Same-origin_policy
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Web/Security")}}
```
Comment (yin1999) [inline files/zh-tw/web/security/same-origin_policy/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Security")}}
```

File: files/zh-tw/web/security/weak_signature_algorithm/index.md [hunk 12]
```diff
@@ -3,6 +3,8 @@ title: 弱簽章演算法
 slug: Web/Security/Weak_Signature_Algorithm
 ---
 
+{{QuickLinksWithSubpages("/en-US/docs/Web/Security")}}
```
Comment (yin1999) [inline files/zh-tw/web/security/weak_signature_algorithm/index.md:6]:
```suggestion
{{QuickLinksWithSubpages("/zh-CN/docs/Web/Security")}}
```

**Review by yin1999 — COMMENTED** [review]
The `{{QuickLinksWithSubpages}}` macro does not work properly, as the `process_array` function inside the [`tree`](https://github.com/mdn/yari/blob/89b79b122efcf7f02e9cf77bb688d2d08d77a5a0/kumascript/src/api/wiki.ts#L119) method it used filter out all documents which do not have the same `locale`.

I think we need a way to collect all the documentation and it could fall back to `en-US`'s docs if localized docs do not exist.

**Review by yin1999 — APPROVED** [review]
Thank you @caugner

**Comment by caugner** [comment]
> The `{{QuickLinksWithSubpages}}` macro does not work properly, as the `process_array` function inside the [`tree`](https://github.com/mdn/yari/blob/89b79b122efcf7f02e9cf77bb688d2d08d77a5a0/kumascript/src/api/wiki.ts#L119) method it used filter out all documents which do not have the same `locale`.
> 
> I think we need a way to collect all the documentation and it could fall back to `en-US`'s docs if localized docs do not exist.

@yin1999 Good catch, I'm hesitant to touch that macro, as @fiji-flo is currently rewriting the build system including macros, which should make that macro in its current form obsolete. However, if this is a trivial fix, like removing the locale condition in that `process_array` function, and if this doesn't break anything else, then I think it would okay to fix before the rewrite lands. Do you want to take a look?

**Comment by yin1999** [comment]
> @yin1999 Good catch, I'm hesitant to touch that macro, as @fiji-flo is currently rewriting the build system including macros, which should make that macro in its current form obsolete. However, if this is a trivial fix, like removing the locale condition in that `process_array` function, and if this doesn't break anything else, then I think it would okay to fix before the rewrite lands. Do you want to take a look?

I've made a try here: mdn/yari#11275

**Pull request merged**